app/models/vm/operations.rb

```diff
@@ -14,7 +14,14 @@ module Vm::Operations
   end
 
   def cockpit_url
-    URI::HTTP.build(:host => ipaddresses.first, :port => 9090).to_s
+    URI::HTTP.build(:host => get_ipv4_address, :port => 9090).to_s
```

Comment:
You call it `get_ipv4_address` first and then `return_ipv4_address`

I am afraid it may not work.